internal/connect/zypper.go

```diff
@@ -66,6 +90,12 @@ func parseProductsXML(xmlDoc []byte) ([]Product, error) {
 	if err := xml.Unmarshal(xmlDoc, &products); err != nil {
 		return []Product{}, err
 	}
+	// override ProductType with OEM value if defined
+	for i, p := range products.Products {
+		if oemValue, err := oemReleaseType(p.ProductLine); err == nil {
+			products.Products[i].ReleaseType = oemValue
+		}
+	}
```

Comment:
sorry, just realizing this now, but maybe it would be better if this was done at the end of `installedProducts()` instead of here. There are some existing tests for `parseProductsXML()` so they may now be invoking `oemReleaseType()` on the test system which is not ideal. Also this does not fit with the name of the function as it is nothing to do with parsing XML.

Reply:
In original this is done in the constructor of `SUSE::Connect::Zypper::Product` so I don't think that running this in tests is a problem.
There are some special cases placed in JSON unmarshaler maybe we should do the same for XML. I know that JSON version doesn't reach out to external resources but still this might be the most logical option.

Reply:
it's not causing a problem now, but ideally tests would not be interacting with the system. I don't think adding this to a custom xml unmarshaler is more logical.

Reply:
we do the same when testing calls which use credentials file. I think we're good as long as we use temp directories and not leave anything behind (which could be picked up by other tests).